src/ui/SurveyUnit/Communication/CommunicationDialogContent.tsx

```diff
@@ -0,0 +1,84 @@
+import {
+  DialogTitle,
+  DialogContent,
+  Box,
+  RadioGroup,
+  Stack,
+  DialogActions,
+  Button,
+} from '@mui/material';
+import { RadioLine } from '../../RadioLine';
+import { useEffect, useState } from 'react';
+import D from '../../../i18n/build-dictionary';
+
+interface CommunicationDialogContentProps {
+  options: {
+    label: string;
+    value: string;
+    disabled: boolean;
+  }[];
+  setCommunicationValue: Function;
+  title: string;
+  lastPickedPropValue: string;
+  isFirst: boolean;
+  nextStep: Function;
+  onClose: Function;
+  previousStep: Function;
+}
+
+const CommunicationDialogContent = ({
+  options,
+  setCommunicationValue,
+  title,
+  lastPickedPropValue,
+  isFirst,
+  nextStep,
+  onClose,
+  previousStep,
+}: CommunicationDialogContentProps) => {
+  const [value, setValue] = useState(
+    options.find(o => o.value === lastPickedPropValue && !o.disabled)
+      ? lastPickedPropValue
+      : options.find(o => !o.disabled)?.value ?? []
+  );
+
+  useEffect(() => {
+    setCommunicationValue(value);
```

Comment:
Why to you need this extra `useEffect` ? Maybe we call `setCommunicationValue` directly from the `onChange`handler (line 55).  And create a local variable instead of the value state. 

```typescript
const value = options.find(o => o.value === lastPickedPropValue && !o.disabled)
      ? lastPickedPropValue
      : options.find(o => !o.disabled)?.value ?? []
```